app/protocol/flows/v4/transactionrelay/handle_relayed_transactions.go

```diff
@@ -50,7 +50,12 @@ func (flow *handleRelayedTransactionsFlow) start() error {
 			return err
 		}
 
-		if flow.IsIBDRunning() {
+		isNearlySynced, err := flow.IsNearlySynced()
```

Comment:
This might be a performance-sensitive location, if we don't optimize `IsNearlySynced` using push updates as suggested above.

Reply:
Maybe it's fine to keep using `IsIBDRunning` in here